### ISSUE TITLE
FKs for self-referencing tables not created

### ISSUE THREAD

**Issue by dsav** [msallin/SQLiteCodeFirst#63]
Foreign keys for self-referencing tables are not created.

For example, have a look at this entity:

``` sql
public class EventInfo
{
    public int Id { get; set; }
    public virtual ICollection<EventInfo> ChildEventInfos { get; set; }
    public int? ParentEventInfoId { get; set; }
    public virtual EventInfo ParentEventInfo { get; set; }
}
```

and this configuration of it:

``` sql
modelBuilder.Entity<EventInfo>()
    .HasOptional(x => x.ParentEventInfo)
    .WithMany(x => x.ChildEventInfos)
    .HasForeignKey(x => x.ParentEventInfoId);
```

Created table lacks FK `ParentEventInfoId REFERENCES EventInfo (Id)`.

The problem seems to be in `CreateDatabaseStatementBuilder.GetCreateTableStatements`, where respective association type is filtered out since its `Constraint.RoleTo.Name` is `"EventInfoSelf"`, not `"EventInfo"`.

**Comment by msallin** [comment]
Will release a new version to fix this bug shortly.
https://github.com/msallin/SQLiteCodeFirst/pull/64

**Comment by msallin** [comment]
Fixed with the release [v1.2.1](https://github.com/msallin/SQLiteCodeFirst/releases/tag/v1.2.1.13)
